### ISSUE TITLE
Dataproc: No functional channel service provider found. Try adding a dependency on the grpc-okhttp, grpc-netty, or grpc-netty-shaded artifact with scala

### ISSUE THREAD

**Issue by deeptibhardwaj24** [googleapis/google-cloud-java#5679]
#### Environment details: 
1. Google cloud dataproc
2. OS type and version: Windows 7
3. Java version:1.8

#### Steps to reproduce

  1. I created **Akka http** API having a route exposed to create a dataproc cluster on GCP although the same cluster creation works fine when google cloud dataproc is used in isolation.

These are dependencies used:

```
libraryDependencies ++= Seq("com.typesafe.akka" %% "akka-actor" % "2.5.12",
  "com.typesafe.akka" %% "akka-http" % "10.1.1",
  "com.typesafe.akka" %% "akka-stream" % "2.5.12",
  "com.typesafe.akka" %% "akka-actor" % "2.5.12" % "runtime",
  "com.typesafe.akka" %% "akka-http" % "10.1.1" % "runtime",
  "com.typesafe.akka" %% "akka-stream" % "2.5.12" % "runtime",
  "org.scalatest" %% "scalatest" % "3.0.3",
  "com.fasterxml.jackson.core" % "jackson-databind" % "2.0.1",
  "org.apache.hive" % "hive-jdbc" % "1.0.0",
  "org.apache.logging.log4j" % "log4j-core" % "2.11.0",
  "org.apache.logging.log4j" % "log4j-api" % "2.11.0",
  "org.apache.hadoop" % "hadoop-client" % "2.6.0",
  "com.typesafe" % "config" % "1.3.1",
  "postgresql" % "postgresql" % "9.1-901-1.jdbc4",
  "ch.qos.logback" % "logback-classic" % "1.2.3",
  "com.typesafe.scala-logging" %% "scala-logging" % "3.5.0",
  "com.jcraft" % "jsch" % "0.1.54",
  "com.typesafe.akka" %% "akka-testkit" % "2.5.12" % Test,
  "org.mockito" % "mockito-core" % "2.11.0" % Test,
  "org.scalaj" %% "scalaj-http" % "2.4.0",
  "com.typesafe.akka" %% "akka-http-testkit" % "10.1.1" % Test,
  "com.google.cloud" % "google-cloud-dataproc" % "0.80.0-alpha")
```
Tried excluding transitive dependencies using exclude and including dependencies mentioned in the above error one at a time, got no luck.
 This is the stack trace:

`No functional channel service provider found. Try adding a dependency on the grpc-okhttp, grpc-netty, or grpc-netty-shaded artifact io.grpc.ManagedChannelProvider$ProviderNotFoundException: No functional channel service provider found. Try adding a dependency on the grpc-okhttp, grpc-netty, or grpc-ne at io.grpc.ManagedChannelProvider.provider(ManagedChannelProvider.java:60) at io.grpc.ManagedChannelBuilder.forAddress(ManagedChannelBuilder.java:37) at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createSingleChannel(InstantiatingGrpcChannelProvider.java:194) at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createChannel(InstantiatingGrpcChannelProvider.java:164) at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.getTransportChannel(InstantiatingGrpcChannelProvider.java:156) at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:157) at com.google.cloud.dataproc.v1beta2.stub.GrpcClusterControllerStub.create(GrpcClusterControllerStub.java:134) at com.google.cloud.dataproc.v1beta2.stub.ClusterControllerStubSettings.createStub(ClusterControllerStubSettings.java:181) at com.google.cloud.dataproc.v1beta2.ClusterControllerClient.(ClusterControllerClient.java:148) at com.google.cloud.dataproc.v1beta2.ClusterControllerClient.create(ClusterControllerClient.java:129) at com.dunnhumby.util.ClusterHelper.getClient(ClusterHelper.scala:21) at com.dunnhumby.util.ClusterHelper.getClient$(ClusterHelper.scala:18) at com.dunnhumby.validation.Validate$.getClient(Validate.scala:34) at com.dunnhumby.util.ClusterHelper.$anonfun$createCluster$1(ClusterHelper.scala:89) at scala.util.Try$.apply(Try.scala:209) at com.dunnhumby.util.ClusterHelper.createCluster(ClusterHelper.scala:81) at com.dunnhumby.util.ClusterHelper.createCluster$(ClusterHelper.scala:81) at com.dunnhumby.validation.Validate$.createCluster(Validate.scala:34) at com.dunnhumby.validation.Validate$.create(Validate.scala:45) at com.dunnhumby.service.RequestHandler.$anonfun$route$25(RequestHandler.scala:67) at akka.http.scaladsl.server.directives.RouteDirectives.$anonfun$complete$1(RouteDirectives.scala:47) at akka.http.scaladsl.server.StandardRoute$$anon$1.apply(StandardRoute.scala:19) at akka.http.scaladsl.server.StandardRoute$$anon$1.apply(StandardRoute.scala:19) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRouteResult$2(BasicDirectives.scala:59) at akka.http.scaladsl.server.directives.FutureDirectives.$anonfun$onComplete$3(FutureDirectives.scala:37) at akka.http.scaladsl.util.FastFuture$.$anonfun$transformWith$1(FastFuture.scala:37) at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41) at akka.http.scaladsl.util.FastFuture$.transformWith$extension1(FastFuture.scala:45) at akka.http.scaladsl.util.FastFuture$.transformWith$extension0(FastFuture.scala:37) at akka.http.scaladsl.server.directives.FutureDirectives.$anonfun$onComplete$2(FutureDirectives.scala:37) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:152) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRouteResult$2(BasicDirectives.scala:59) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:152) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRequestContext$2(BasicDirectives.scala:41) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:152) at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$2(RouteConcatenation.scala:47) at akka.http.scaladsl.util.FastFuture$.strictTransform$1(FastFuture.scala:41) at akka.http.scaladsl.util.FastFuture$.transformWith$extension1(FastFuture.scala:45) at akka.http.scaladsl.util.FastFuture$.flatMap$extension(FastFuture.scala:26) at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$1(RouteConcatenation.scala:44) at akka.http.scaladsl.server.RouteConcatenation$RouteWithConcatenation.$anonfun$$tilde$1(RouteConcatenation.scala:44) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$mapRouteResultWith$2(BasicDirectives.scala:65) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:152) at akka.http.scaladsl.server.directives.ExecutionDirectives.$anonfun$handleExceptions$2(ExecutionDirectives.scala:32) at akka.http.scaladsl.server.directives.BasicDirectives.$anonfun$textract$2(BasicDirectives.scala:152) at akka.http.scaladsl.server.Route$.$anonfun$asyncHandler$1(Route.scala:86) at akka.stream.impl.fusing.MapAsync$$anon$25.onPush(Ops.scala:1190) at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:519) at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482) at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378) at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:588) at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:472) at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:563) at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:745) at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:760) at akka.actor.Actor.aroundReceive(Actor.scala:517) at akka.actor.Actor.aroundReceive$(Actor.scala:515) at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:670) at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588) at akka.actor.ActorCell.invoke(ActorCell.scala:557) at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258) at akka.dispatch.Mailbox.run(Mailbox.scala:225) at akka.dispatch.Mailbox.exec(Mailbox.scala:235) at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) `

**Comment by deeptibhardwaj24** [comment]
Closing the issue as there was an issue with the MergeStrategy for META-INF in build.sbt. It works fine after applying MergeStrategy.discard.

**Comment by hmallepally** [comment]
may not be relevant but it is possible my response can help someone save time

I found this to be a dependency issue. I have seen this problem and upon investigation, the code is internally trying to load android Application class with class.forName and if successful, it treats it as android and otherwise it is non android.

problem I saw was because some dependency I had android jar into my classpath (google guava android or so) so excluded that dependency and it worked pretty fine.

I think the error message should be improved and give more meaningful message than asking to add grpc-netty, because the error can happen even with grpc-netty in classpath.
